velox/vector/FlatVector.h

```diff
@@ -403,6 +403,12 @@ class FlatVector final : public SimpleVector<T> {
     return nullptr;
   }
 
+  // Allocate buffer for not inline StringView from stringBuffers or pool,
```

Comment:
It would be nice to document the existing getBufferWithSpace method and make it easy for the reader to understand the difference between the 2 methods.

> Allocate buffer ... from stringBuffers or pool

Buffers can only be allocated from the pool, hence, it is confusing to say "allocate from stringBuffers." Maybe,

```
Finds an existing string buffer that's singly-referenced (not shared) and have enough unused capacity to fit 'size' bytes. If found, resizes the buffer to add 'size' bytes and returns a pointer to the start of writable memory. If not found, allocates new buffer, adds it to 'stringBuffers', sets buffer size to 'size' and returns a pointer to the start of writable memory.

The caller needs to make sure not to write more then 'size' bytes.
```